db/db.go

```diff
@@ -120,21 +124,22 @@
     contract             VARCHAR(255), 
     ref                  VARCHAR, 
     toAddr               VARCHAR(255),
-    amount               VARCHAR
+	amount               VARCHAR,
+	tag0                 CHAR(64)
 );`
 	_, err = db.db.Exec(shiftOut)
 	return err
 }
 
 // InsertTx implements the `DB` interface.
-func (db database) InsertTx(tx abi.Tx, gaas bool) error {
+func (db database) InsertTx(tx abi.Tx, tag string, gaas bool) error {
```

Comment:
What encoding is the tag using? We should be able to put an upper limit of the length of the tag, and thus be able to enforce that limit here (as a defensive check).